app/controllers/api/courses/github_repos_controller.rb

```diff
@@ -14,11 +15,22 @@ def index
         unless visibility_query.nil? || visibility_query.empty?
           @github_repos = @github_repos.where(visibility: visibility_query)
         end
-        paginate json: @github_repos
+
+        github_repos_array = @github_repos.to_a
+        github_repos_hash_array = github_repos_array.map{ |repo| repo_to_hash(repo) }
+        paginate json: github_repos_hash_array
```

Comment:
This will be really slow because you're making a separate db query in `#repo_to_hash` for each individual repo (N+1 query problem), particularly since this operation is being performed on all the repo records for a course, not just the current page vis-a-vis JSON pagination. I would *strongly* recommend using either [a counter cache](https://stackoverflow.com/a/9209705/3950780) or [an include helper method](https://stackoverflow.com/a/9371631/3950780) to implement this behavior so that you can fetch these counts all in a single query.

Reply:
For the time being, the performance problem isn't an issue.   Make it work, make it better, make it fast.   We are at the "make it work" stage.

Reply:
However, I'll put this in an issue so that we come back to it.

Reply:
Revisit in issue #299